### PR TITLE
Updated aya_vision.md

### PR DESCRIPTION
**Pull request by 1himan**
# What does this PR do?

As a part of this - https://github.com/huggingface/transformers/issues/36979

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/aya_vision.md [hunk 1]
```diff
@@ -14,82 +14,30 @@ rendered properly in your Markdown viewer.
...
-- High-quality visual understanding using the Siglip2-so400-384-14 vision encoder
-- Seamless integration of visual and textual information in 23 languages.
-
-<!-- <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/aya_vision_architecture.webp"
-alt="drawing" width="600"/>
-
-<small> Aya Vision architecture. </small> -->
-
-Tips:
+<div style="float: right;">
```
Comment (stevhliu) [inline docs/source/en/model_doc/aya_vision.md:17]: Nice!

File: docs/source/en/model_doc/aya_vision.md [hunk 2]
```diff
@@ -14,82 +14,30 @@ rendered properly in your Markdown viewer.
...
-
-
-## Usage
-
-Here's how to use Aya Vision for inference:
-
-```python
-from transformers import AutoProcessor, AutoModelForImageTextToText
-import torch
+# AyaVision
```
Comment (stevhliu) [inline docs/source/en/model_doc/aya_vision.md]:
```suggestion
# Aya Vision
```

File: docs/source/en/model_doc/aya_vision.md [hunk 3]
```diff
@@ -14,82 +14,30 @@ rendered properly in your Markdown viewer.
...
 
-model_id = "CohereForAI/aya-vision-8b"
-torch_device = "cuda:0"
+[AyaVision](https://huggingface.co/papers/2505.08751) is a family of open-weight multimodal vision-language models from Cohere Labs. Each model integrates a Command R7B–based multilingual transformer that has been post-trained with a synthetic multimodal instruction pipeline, together with a SigLIP2-patched vision encoder connected via lightweight adapter layers. The system processes images as tiled 364 × 364 patches plus a low-resolution thumbnail and supports up to 16 K tokens of text. It performs OCR, image captioning, visual reasoning, multilingual question answering in 23 languages, and code-comment generation without compromising text-only tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/aya_vision.md]:
```suggestion
[Aya Vision](https://huggingface.co/papers/2505.08751) is a family of open-weight multimodal vision-language models from Cohere Labs. It is trained with a synthetic annotation framework that generates high-quality multilingual image captions, improving Aya Vision's generated responses. In addition, a cross-modal model merging technique is used to prevent the model from losing its text capabilities after adding vision capabilities. The model combines a CommandR-7B language model with a SigLIP vision encoder.
```

File: docs/source/en/model_doc/aya_vision.md [hunk 4]
```diff
@@ -14,82 +14,30 @@ rendered properly in your Markdown viewer.
...
-
-# Format message with the aya-vision chat template
-messages = [
-    {"role": "user",
-     "content": [
-       {"type": "image", "url": "https://pbs.twimg.com/media/Fx7YvfQWYAIp6rZ?format=jpg&name=medium"},
-        {"type": "text", "text": "चित्र में लिखा पाठ क्या कहता है?"},
-    ]},
-    ]
+You can find all the original AyaVision checkpoints under the [AyaVision](https://huggingface.co/collections/CohereLabs/cohere-labs-aya-vision-67c4ccd395ca064308ee1484) collection.
```
Comment (stevhliu) [inline docs/source/en/model_doc/aya_vision.md]:
```suggestion
You can find all the original Aya Vision checkpoints under the [Aya Vision](https://huggingface.co/collections/CohereLabs/cohere-labs-aya-vision-67c4ccd395ca064308ee1484) collection.
```

File: docs/source/en/model_doc/aya_vision.md [hunk 5]
```diff
@@ -14,82 +14,30 @@ rendered properly in your Markdown viewer.
...
 
-# Process image on CUDA
-inputs = processor.apply_chat_template(
-    messages, padding=True, add_generation_prompt=True, tokenize=True, return_dict=True, return_tensors="pt", device=torch_device
-).to(model.device)
+> [!TIP]
+> Click on the AyaVision models in the right sidebar for more examples of how to apply AyaVision to different image-to-text tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/aya_vision.md]:
```suggestion
> Click on the Aya Vision models in the right sidebar for more examples of how to apply Aya Vision to different image-to-text tasks.
```

File: docs/source/en/model_doc/aya_vision.md [hunk 6]
```diff
@@ -102,132 +50,61 @@ messages = [
...
-For processing batched inputs (multiple conversations at once):
+</hfoption>
 
-```python
-from transformers import AutoProcessor, AutoModelForImageTextToText
-import torch
+</hfoptions>
 
-model_id = "CohereForAI/aya-vision-8b"
+Quantization reduces the memory footprint of large models by representing weights at lower precision. Refer to the [Quantization](https://huggingface.co/docs/transformers/main/en/main_classes/quantization) overview for supported backends.
```
Comment (stevhliu) [inline docs/source/en/model_doc/aya_vision.md]: Add a code example of quantization with perhaps a larger checkpoint

```suggestion
Quantization reduces the memory footprint of large models by representing weights at lower precision. Refer to the [Quantization](../quantization/overview) overview for supported backends.
```

File: docs/source/en/model_doc/aya_vision.md [hunk 7]
```diff
@@ -102,132 +50,61 @@ messages = [
...
-# Process each conversation separately and combine into a batch
-batch_inputs = processor.apply_chat_template(
-    batch_messages, 
-    padding=True, 
-    add_generation_prompt=True, 
-    tokenize=True, 
-    return_dict=True, 
-    return_tensors="pt"
-).to(model.device)
+- Use explicit language tokens (e.g. `<en>`, `<fr>`, `<hi>`) in prompts to reliably set the output language—otherwise it infers from input, which may be less precise.
```
Comment (stevhliu) [inline docs/source/en/model_doc/aya_vision.md]:
```suggestion
- Images are represented with the `<image>` tag in the chat template.
```

File: docs/source/en/model_doc/aya_vision.md [hunk 8]
```diff
@@ -102,132 +50,61 @@ messages = [
...
 
-# Generate responses for the batch
-batch_outputs = model.generate(
-    **batch_inputs,
-    max_new_tokens=300,
-    do_sample=True,
-    temperature=0.3,
-)
+- You can pass a list of messages to the `pipeline` or manually format them via `apply_chat_template`, allowing multiple images or prompts in a single forward pass.
```
Comment (stevhliu) [inline docs/source/en/model_doc/aya_vision.md]:
```suggestion
- Use the [`~ProcessorMixin.apply_chat_template`] method to correctly format inputs.
```

File: docs/source/en/model_doc/aya_vision.md [hunk 9]
```diff
@@ -102,132 +50,61 @@ messages = [
...
 
-# Decode the generated responses
-for i, output in enumerate(batch_outputs):
-    response = processor.tokenizer.decode(
-        output[batch_inputs.input_ids.shape[1]:], 
-        skip_special_tokens=True
-    )
-    print(f"Response {i+1}:\n{response}\n")
-```
+- The model uses `use_cache=True` by default, so repeated `generate(...)` calls retain past key/value states and speed up decoding for long outputs.
```
Comment (stevhliu) [inline docs/source/en/model_doc/aya_vision.md]: Add code snippet from [here](https://huggingface.co/docs/transformers/main/en/model_doc/aya_vision#multiple-images-and-batched-inputs)

```suggestion
- The example below demonstrates inference with multiple images.

   ```py
   ```
```

Comment (stevhliu) [inline docs/source/en/model_doc/aya_vision.md]: Add [code snippet](https://huggingface.co/docs/transformers/main/en/model_doc/aya_vision#multiple-images-and-batched-inputs) for batched inputs too

```md
The example below demonstrates inference with batched inputs.

   ```py
   add code snippet here
```

File: docs/source/en/model_doc/aya_vision.md [hunk 10]
```diff
@@ -243,5 +120,5 @@ for i, output in enumerate(batch_outputs):
 
 ## AyaVisionForConditionalGeneration
 
-[[autodoc]] AyaVisionForConditionalGeneration
+[[autodoc]] AyaVisionForConditionalGeneration
```
Comment (stevhliu) [inline docs/source/en/model_doc/aya_vision.md]:
```suggestion
[[autodoc]] AyaVisionForConditionalGeneration
```

**Review by stevhliu — COMMENTED** [review]
Nice work, thank you for this!

File: docs/source/en/model_doc/aya_vision.md [hunk 11]
```diff
@@ -70,118 +77,93 @@ messages = [
...
-print(outputs)
 ```
 
-### Multiple Images and Batched Inputs
+</hfoption>
 
-Aya Vision can process multiple images in a single conversation. Here's how to use it with multiple images:
+</hfoptions>
 
+Quantization reduces the memory footprint of large models by representing weights at lower precision. Refer to the [Quantization](https://huggingface.co/docs/transformers/v4.52.3/quantization/overview) overview for supported backends.
```
Comment (stevhliu) [inline docs/source/en/model_doc/aya_vision.md]:
```suggestion
Quantization reduces the memory footprint of large models by representing weights at lower precision. Refer to the [Quantization](../quantization/overview) overview for supported backends.

The example below uses [bitsandbytes](../quantization/bitsandbytes) to only quantize the weights to 4-bits.
```

File: docs/source/en/model_doc/aya_vision.md [hunk 12]
```diff
@@ -199,49 +181,108 @@ batch_messages = [
...
+    for i, output in enumerate(batch_outputs):
+        response = processor.tokenizer.decode(
+            output[batch_inputs.input_ids.shape[1]:], 
+            skip_special_tokens=True
+        )
+        print(f"Response {i+1}:\n{response}\n")
+    ```
 
-## AyaVisionProcessor
+## Aya VisionProcessor
```
Comment (stevhliu) [inline docs/source/en/model_doc/aya_vision.md]: You don't need to change any of these :)

They should still be `## AyaVisionProcessor` without a space in between

**Review by stevhliu — APPROVED** [review]
Just a few more minor comments and then we can merge!

File: docs/source/en/model_doc/aya_vision.md (hunk 12 above)
Comment (1himan) [reply docs/source/en/model_doc/aya_vision.md]: Done

File: docs/source/en/model_doc/aya_vision.md (hunk 11 above)
Comment (1himan) [reply docs/source/en/model_doc/aya_vision.md]: 👍
